pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,25 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+// An interface for types that allow getting information about a Node on which give pod is running.
+type HostLocator interface {
```

Comment:
sorry for commenting late, only saw this PR when it merged... I think the `ConnectionInfoGetter` interface that is already responsible for returning the kubelet port should be updated to provide this info...